### PR TITLE
feat(core): support targets with colons in the name without quotes

### PR DESCRIPTION
**Pull request by AgentEnder**
See #13914

### CODE REVIEW

File: packages/nx/src/utils/split-target.spec.ts [hunk 1]
```diff
@@ -1,14 +1,40 @@
+import { ProjectGraphProjectNode } from '@nrwl/nx-cloud/lib/core/models/run-context.model';
```
Comment (FrozenPandaz) [inline packages/nx/src/utils/split-target.spec.ts:1]: Wrong import

File: packages/react/src/generators/cypress-component-configuration/lib/add-files.ts [hunk 2]
```diff
@@ -81,7 +82,10 @@ function getBundler(found: FoundTarget, tree: Tree): 'vite' | 'webpack' {
     return found.config.executor === '@nrwl/vite:build' ? 'vite' : 'webpack';
   }
 
-  const { target, project } = parseTargetString(found.target);
+  const { target, project } = parseTargetString(
+    found.target,
+    readCachedProjectGraph()
```
Comment (FrozenPandaz) [inline packages/react/src/generators/cypress-component-configuration/lib/add-files.ts:87]:
```suggestion
    await createProjectGraphAsync()
```

File: packages/storybook/src/generators/configuration/util-functions.ts [hunk 3]
```diff
@@ -551,7 +551,8 @@ export function getE2EProjectName(
       }
       if (options['devServerTarget']) {
         const { project, target } = parseTargetString(
-          options['devServerTarget']
+          options['devServerTarget'],
+          readCachedProjectGraph()
```
Comment (FrozenPandaz) [inline packages/storybook/src/generators/configuration/util-functions.ts:555]:
```suggestion
    await createProjectGraphAsync()
```

File: packages/workspace/src/generators/remove/lib/check-targets.ts [hunk 4]
```diff
@@ -27,7 +28,10 @@ export function checkTargets(tree: Tree, schema: Schema) {
     Object.entries(projectConfig.targets || {}).forEach(([, targetConfig]) => {
       checkIfProjectIsUsed(targetConfig, (value) => {
         try {
-          const { project } = parseTargetString(value);
+          const { project } = parseTargetString(
+            value,
+            readCachedProjectGraph()
```
Comment (FrozenPandaz) [inline packages/workspace/src/generators/remove/lib/check-targets.ts:33]:
```suggestion
    await createProjectGraphAsync()
```

**Review by FrozenPandaz — DISMISSED** [review]

File: packages/angular/src/builders/webpack-dev-server/webpack-dev-server.impl.ts [hunk 5]
```diff
@@ -26,7 +30,10 @@ export function executeWebpackDevServerBuilder(
 
   const options = normalizeOptions(rawOptions);
 
-  const parsedBrowserTarget = parseTargetString(options.browserTarget);
+  const parsedBrowserTarget = parseTargetString(
+    options.browserTarget,
+    readCachedProjectGraph()
```
[the remark this answers is not in the document]
Comment (AgentEnder) [reply packages/angular/src/builders/webpack-dev-server/webpack-dev-server.impl.ts:35]: This doesn't exist in this one... Its an angular builder.

File: docs/generated/devkit/index.md [hunk 6]
```diff
@@ -134,7 +134,11 @@ It only uses language primitives and immutable objects
 - [output](../../devkit/documents/index#output)
 - [workspaceRoot](../../devkit/documents/index#workspaceroot)
 
-### Functions
+### Executors Functions
```
[the remark this answers is not in the document]
Comment (AgentEnder) [reply docs/generated/devkit/index.md:137]: I have zero idea what's causing the doc generator to create this category :/

File: packages/devkit/src/executors/parse-target-string.ts [hunk 7]
```diff
@@ -1,19 +1,35 @@
...
  */
-export function parseTargetString(targetString: string): Target {
-  const [project, target, configuration] = splitTarget(targetString);
+export function parseTargetString(
+  targetString: string,
+  projectGraph: ProjectGraph
+): Target;
+export function parseTargetString(
+  targetString: string,
+  projectGraph = readCachedProjectGraph()
```
[the remark this answers is not in the document]
Comment (AgentEnder) [reply packages/devkit/src/executors/parse-target-string.ts:27]: Typescript overrides have to have one signature which accepts all of the overrides.

File: packages/nx/src/utils/split-target.spec.ts (hunk 1 above)
Comment (AgentEnder) [reply packages/nx/src/utils/split-target.spec.ts:1]: Oh, wow. yeah, thanks.

File: packages/devkit/src/executors/parse-target-string.spec.ts [hunk 8]
```diff
@@ -14,7 +16,10 @@ const cases = [
 
 describe('parseTargetString', () => {
   it.each(cases)('$input -> $expected', ({ input, expected }) => {
-    expect(parseTargetString(input)).toEqual(expected);
+    jest
```
[the remark this answers is not in the document]
Comment (AgentEnder) [reply packages/devkit/src/executors/parse-target-string.spec.ts:19]: This mock makes it s.t. these tests function as a test for parse-target-string. The splitTarget function is tested separately.

File: packages/nx/src/project-graph/project-graph-builder.ts [hunk 9]
```diff
@@ -44,6 +44,7 @@ export class ProjectGraphBuilder {
     }
     this.graph.nodes[node.name] = node;
     this.graph.dependencies[node.name] = [];
+    return this;
```
[the remark this answers is not in the document]
Comment (AgentEnder) [reply packages/nx/src/project-graph/project-graph-builder.ts:47]: Sure, I can drop it. This change makes it a lot easier to use the builder declaratively, but that is mostly useful in tests.

File: docs/generated/devkit/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (AgentEnder) [reply docs/generated/devkit/index.md:137]: Never mind. Found it, didn't change it so not really sure why this changed.

File: packages/nx/src/project-graph/project-graph-builder.ts [hunk 10]
```diff
@@ -28,7 +28,7 @@ export class ProjectGraphBuilder {
   /**
    * Adds a project node to the project graph
    */
-  addNode(node: ProjectGraphProjectNode): void {
+  addNode<T extends ProjectGraphProjectNode>(node: T) {
```
Comment (vsavkin) [inline packages/nx/src/project-graph/project-graph-builder.ts:31]: Why is this required ?

File: packages/nx/src/utils/split-target.spec.ts [hunk 11]
```diff
@@ -1,14 +1,44 @@
+import { ProjectGraph, ProjectGraphProjectNode } from '../config/project-graph';
+import { ProjectConfiguration } from '../config/workspace-json-project-json';
+import { ProjectGraphBuilder } from '../project-graph/project-graph-builder';
 import { splitTarget } from './split-target';
 
 const cases = [
```
Comment (vsavkin) [inline packages/nx/src/utils/split-target.spec.ts:6]: remove the loop. inline everything.

**Review by vsavkin — APPROVED** [review]
The change is good. Two small things:

- We should not change the interface of a core object just for tests.
- Loops in tests is a bad idea.
